packages/ove-asset-manager/README.md

```diff
@@ -1,7 +1,112 @@
+
+The Asset Processing Microservices work on a pull based model - they will each independently regularly request further work. You are encouraged to run multiple copies of each Asset Processing Microservice: these should be configured to communicate with an instance of the Asset Manager. These services will then periodically check for Assets to process. Your database and object store may each scale independently. 
+
+## Storage Model
+
+Assets will be stored on the object store on a one bucket per `Project` model. Every Asset is assigned a ``GUID`` on creation. This is used to identify it in the SQL database and on the object store. Within the Project bucket each Asset is uploaded to the root of a folder named by this ``GUID``. The filename of the asset will be preserved where possible - file names must consist of `a-z`,`A-Z`, `()`, `-` or `_` and be of length less than `1024` characters. Asset processing services are free to place processed files and folders anywhere within the asset directory. 
+
+## Configuration
+
+Configuration of the service is achieved by modifying the ``appsettings.json`` file. These settings are automatically over written by **Environment Variables** (reflection of the JSON hierarcy should be acheived using ``__`` e.g. ``s3Client__Secret``). 
+
+### Configuring S3
+
+In common with all S3 compatible object stores the following three properties are required and should be set as follows in ``appsettings.json`` or overridden by ``environment variables`` as discussed above.
+
+```  
+"s3Client": {
+    "AccessKey": "key",
     "Secret": "secret",
     "ServiceURL": "host"
-  },
\ No newline at end of file
+  }
+ ```
+ 
+### Configuring Maria DB
+The [MariaDB](https://mariadb.org/) configuration should be set as follows in `appsettings.json` or overridden by ``Environment Variables`` as discussed above. 
+
+```
+  "MariaDB": { 
+    "ConnectionString": "Server=SERVER;Port=3306;Database=DATABASE;User=USERNAME;Password=PASSWORD;", 
+    "Version" :  "10.3.10"  
+  }
+```
+
+## Using the Asset Manager
+
+Assets should be uploaded to the asset manager either through the web interface or submitting a HTTP `POST` request to the `/OVEAssetModelController/Create/` API. 
+Asset URLs on the object store may be found via the `/OVEAssetModelController/GetAssetURL?project=X&file=Y` API or by ID on  `/OVEAssetModelController/GetAssetURLbyId?id=X`
+
+## Versioning
+
+The Asset Manager supports file versioning. The service will, unless otherwise specified, always provide the most recently uploaded/edited asset version when searching by Project and Asset Name. Previous versions may be found via the `/OVEAssetModelController/ListAssets/Project/` and `/OVEAssetModelController/ListAssets/Project/Name/` API's.

```

Comment:
```suggestion
The Asset Manager supports file versioning. The service will, unless otherwise specified, always provide the most recently uploaded/edited asset version when searching by Project and Asset Name. Previous versions may be found via the `/OVEAssetModelController/ListAssets/Project/` and `/OVEAssetModelController/ListAssets/Project/Name/` API routes.
```